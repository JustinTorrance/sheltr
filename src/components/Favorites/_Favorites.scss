.change-view {
  border-radius: 20px;
  background-color: black;
  font-family: 'Buda', cursive;
  color: white;
  font-size: 1.3rem;
  height: 60px;
  width: 150px;
  position: absolute;
  right: 280px;
   &:hover {
    cursor: pointer;
    color: black;
    background-color: white;
    transition: .2s;
   }
}

.shelter-name {
  text-align: center;
}