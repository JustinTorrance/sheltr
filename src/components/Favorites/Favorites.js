import React from 'react'
import { connect } from 'react-redux'
import './_Favorites.scss'
import { addFavorite, removeFavorite } from '../../actions'
import Card from '../Card/Card.js'
import { Link } from 'react-router-dom'

export const Favorites = (props) => {

  return (
    <section className='pets-results'>
      <Link to='/display'>
        <button>Back to Shelter</button>
      </Link>
      <h3 className='favorites-name'>Favorites</h3>
      <section className='pets-grid'>
        {
          props.favorites.map(favorite => {
            return (
              <Card {...favorite} />
            )            
          })          
        }
      </section>
    </section>
  )
}

export const mapStateToProps = (state) => ({
  petsFromShelter: state.petsFromShelter,
  favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
  addFavorite: (pet) => dispatch(addFavorite(pet)),
  removeFavorite: (pet) => dispatch(removeFavorite(pet))
})

export default connect(mapStateToProps, mapDispatchToProps)(Favorites)