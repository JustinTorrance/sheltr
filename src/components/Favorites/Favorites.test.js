import { mapStateToProps } from './Favorites'
import React from 'react';
import Favorites from './Favorites'

describe('Favorites', () => {
  it('matches the snapshot', () => {
    const wrapper = <Favorites />
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an array favorite pets', () => {
      const mockState = {
        favorites: [{name: 'fluffy'}]
      }
      const expected = {
        favorites: [{name: 'fluffy'}] 
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a addFavorite thunk when handeSubmit is called', () => {
      const mockDispatch = jest.fn()
      const thunkToDispatch = addFavorite(e)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleFavorite(e)
      expect(mockDispatch).toHaveBeenCalledWith(thunkToDispatch)
    })
    it('calls dispatch with a removeFavorite thunk when handeSubmit is called', () => {
      const mockDispatch = jest.fn()
      const thunkToDispatch = removeFavorite(e)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleFavorite(e)
      expect(mockDispatch).toHaveBeenCalledWith(thunkToDispatch)
    })
  })
})