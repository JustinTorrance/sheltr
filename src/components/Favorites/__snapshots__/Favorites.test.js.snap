// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites matches the snapshot 1`] = `<Connect(Favorites) />`;
